### PR TITLE
Add a flag to allow objective modes to be advanced

### PR DESCRIPTION
**Pull request by PotatoStealer**

### CODE REVIEW

File: PGM/src/main/java/tc/oc/pgm/modes/ObjectiveModeCommands.java [hunk 1]
```diff
@@ -69,17 +73,24 @@ private BaseComponent formatMode(ObjectiveMode mode, Duration time) {
...
         aliases = "next",
-        desc = "Shows information about the next mode"
+        flags = "f",
+        desc = "Shows information about the next mode, or advance to the next mode with -f"
     )
     public void next(CommandContext args, CommandSender sender) throws CommandException {
         final ObjectiveModeManager manager = this.manager.get();
         final Audience audience = audiences.get(sender);
         final Optional<Pair<ObjectiveMode, Duration>> next = manager.nextMode();
+        final boolean advance = args.hasFlag('f');
```
Comment (PotatoStealer) [inline PGM/src/main/java/tc/oc/pgm/modes/ObjectiveModeCommands.java]: Probably don't need this boolean here

File: PGM/src/main/java/tc/oc/pgm/modes/ObjectiveModeCommands.java [hunk 2]
```diff
@@ -15,17 +15,21 @@
...
 import tc.oc.commons.core.commands.Commands;
 import tc.oc.commons.core.commands.NestedCommands;
+import tc.oc.commons.core.commands.TranslatableCommandException;
 import tc.oc.commons.core.formatting.PeriodFormats;
 import tc.oc.commons.core.util.Pair;
+import tc.oc.pgm.commands.CommandUtils;
 
 @Singleton
 public class ObjectiveModeCommands implements NestedCommands {
+    public static Permission ADVANCE_PERMISSION = new Permission("pgm.modes.advance");
```
Comment (PotatoStealer) [inline PGM/src/main/java/tc/oc/pgm/modes/ObjectiveModeCommands.java]: static final

File: Commons/core/src/main/i18n/templates/pgm/PGMErrors.properties [hunk 3]
```diff
@@ -23,6 +23,8 @@ command.class.restricted = The {0} class is restricted
 command.class.stickyClass = You may not change classes because your current class is sticky.
 command.class.notEnabled = Classes are not enabled on this map.
 
+command.modes.advance.matchNotStarted = Objective modes may not be forced push until the match starts.
```
Comment (PotatoStealer) [inline Commons/core/src/main/i18n/templates/pgm/PGMErrors.properties]: edit the grammar of the string

**Comment by Pablete1234** [comment]
how does `-f` represent "advance" ? try `-a` ? or something else that would fit

**Comment by Matic0B** [comment]
Force possibly

**Comment by PotatoStealer** [comment]
yeah, i kept it just to keep it uniform with the other commands, i.e. `cycle -f`

**Pull request merged**